static/app/components/globalDrawer/index.tsx

```diff
@@ -0,0 +1,95 @@
+import {createContext, useCallback, useContext, useEffect, useRef, useState} from 'react';
+
+import {DrawerBody, DrawerPanel} from 'sentry/components/globalDrawer/components';
+import type {
+  DrawerConfig,
+  DrawerContext as TDrawerContext,
+} from 'sentry/components/globalDrawer/types';
+import {useHotkeys} from 'sentry/utils/useHotkeys';
+import {useLocation} from 'sentry/utils/useLocation';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+const DEFAULT_DRAWER_CONFIG: DrawerConfig = {
+  renderer: null,
+  options: {
+    closeOnEscapeKeypress: true,
+    closeOnOutsideClick: true,
+  },
+};
```

Comment:
Do we even need this default, what if it was just `undefined` when not set and you just had

```
  const [drawerConfig, setDrawerConfig] = useState<DrawerConfig>();
```

Then instead of checking if the render is set you can just check if the config is not undefined. Should clean up a bit more code